avm/res/cache/redis/main.bicep

```
@@ -0,0 +1,422 @@
+param name string
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. The managed identity definition for this resource.')
+param managedIdentities managedIdentitiesType
+
+@description('Optional. Specifies whether the non-ssl Redis server port (6379) is enabled.')
+param enableNonSslPort bool = false
+
+@allowed([
+  '1.0'
+  '1.1'
+  '1.2'
+])
+@description('Optional. Requires clients to use a specified TLS version (or higher) to connect.')
+param minimumTlsVersion string = '1.2'
+
+@description('Optional. Whether or not public network access is allowed for this resource. For security reasons it should be disabled. If not specified, it will be disabled by default if private endpoints are set.')
+@allowed([
+  ''
+  'Enabled'
+  'Disabled'
+])
+param publicNetworkAccess string = ''
+
+@description('Optional. All Redis Settings. Few possible keys: rdb-backup-enabled,rdb-storage-connection-string,rdb-backup-frequency,maxmemory-delta,maxmemory-policy,notify-keyspace-events,maxmemory-samples,slowlog-log-slower-than,slowlog-max-len,list-max-ziplist-entries,list-max-ziplist-value,hash-max-ziplist-entries,hash-max-ziplist-value,set-max-intset-entries,zset-max-ziplist-entries,zset-max-ziplist-value etc.')
+param redisConfiguration object = {}
+
+@allowed([
+  '4'
+  '6'
+])
+@description('Optional. Redis version. Only major version will be used in PUT/PATCH request with current valid values: (4, 6).')
+param redisVersion string = '6'
+
+@minValue(1)
+@description('Optional. The number of replicas to be created per primary.')
+param replicasPerMaster int = 1
+
+@minValue(1)
+@description('Optional. The number of replicas to be created per primary.')
+param replicasPerPrimary int = 1
+
+@minValue(1)
+@description('Optional. The number of shards to be created on a Premium Cluster Cache.')
+param shardCount int = 1
+
+@allowed([
+  0
+  1
+  2
+  3
+  4
+  5
+  6
+])
+@description('Optional. The size of the Redis cache to deploy. Valid values: for C (Basic/Standard) family (0, 1, 2, 3, 4, 5, 6), for P (Premium) family (1, 2, 3, 4).')
+param capacity int = 1
+
+@allowed([
+  'Basic'
+  'Premium'
+  'Standard'
+])
+@description('Optional. The type of Redis cache to deploy.')
+param skuName string = 'Basic'
+
+@description('Optional. Static IP address. Optionally, may be specified when deploying a Redis cache inside an existing Azure Virtual Network; auto assigned by default.')
+param staticIP string = ''
+
+@description('Optional. The full resource ID of a subnet in a virtual network to deploy the Redis cache in. Example format: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/Microsoft.{Network|ClassicNetwork}/VirtualNetworks/vnet1/subnets/subnet1.')
+param subnetId string = ''
```

```suggestion
@description('Optional. The full resource ID of a subnet in a virtual network to deploy the Redis cache in.')
param subnetResourceId string = ''
```

Note: Needs updating in other locations & regenerations of files